### PR TITLE
feat(options): add multiple `openPage` support

### PR DESCRIPTION
**Pull request by digitaljohn**
- [ ] This is a **bugfix**
- [x] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

Yes and added an example.

### Motivation / Use-Case

Closes issue #2265 

### Breaking Changes

No breaking changes

### Additional Info

Added the ability to specify multiple windows to open using the existing `openPage` option. An array can now be set in the `devServer` Webpack config object or if using the `--open-page` CLI option a comma-delimited-list can be specified.

> Please note regular strings still function correctly. No Breaking Changes.

API:
```js
devServer: {
  openPage: ['example1.html', 'example2.html'],
},
```
CLI:
```bash
npm run webpack-dev-server -- --open-page example1.html,example2.html
```

### CODE REVIEW

File: test/server/utils/runOpen.test.js [hunk 1]
```diff
@@ -45,6 +45,48 @@ describe('runOpen util', () => {
...
+        expect(opn.mock.calls[1]).toMatchInlineSnapshot(`
+          Array [
+            "https://example.com/index2.html",
+            Object {
+              "wait": false,
+            },
+          ]
+        `);
+      });
+    });
```
Comment (evilebottnawi) [inline test/server/utils/runOpen.test.js]: don't use `toMatchInlineSnapshot`, use `toMatchSnapshot`, it is still unstable in some cases

File: examples/cli/open-page/README.md [hunk 2]
```diff
@@ -10,5 +10,5 @@ as the argument to the `open-page` option.
 
 ## What Should Happen
 
-The script should open `http://localhost:8080/` in your default browser.
-You should see the text on the page itself change to read `Success!`.
+The script should open `http://localhost:8080/example.html#page1` in your
+default browser. You should see the text on the page itself change to read `Success!`.
```
Comment (evilebottnawi) [inline examples/cli/open-page/README.md]: I think better create new example, no need modify current

**Comment by evilebottnawi** [comment]
Also please accept CLA

File: examples/cli/open-page/README.md (hunk 2 above)
Comment (digitaljohn) [reply examples/cli/open-page/README.md]: I did create a new example. I just spotted a typo in yours and fixed it?

Comment (evilebottnawi) [reply examples/cli/open-page/README.md]: oh, i see, thanks! All good

File: test/server/utils/runOpen.test.js (hunk 1 above)
Comment (digitaljohn) [reply test/server/utils/runOpen.test.js]: Done

File: examples/cli/open-page/README.md [hunk 3]
```diff
@@ -10,5 +10,5 @@ as the argument to the `open-page` option.
 
 ## What Should Happen
 
-The script should open `http://localhost:8080/` in your default browser.
-You should see the text on the page itself change to read `Success!`.
+The script should open `http://localhost:8080/example.html#page1` in your
```
Comment (hiroppy) [inline examples/cli/open-page/README.md]: 👍

File: lib/options.json [hunk 4]
```diff
@@ -436,7 +447,7 @@
       "noInfo": "should be {Boolean} (https://webpack.js.org/configuration/dev-server/#devservernoinfo-)",
       "onListening": "should be {Function} (https://webpack.js.org/configuration/dev-server/#onlistening)",
       "open": "should be {String|Boolean} (https://webpack.js.org/configuration/dev-server/#devserveropen)",
-      "openPage": "should be {String} (https://webpack.js.org/configuration/dev-server/#devserveropenpage)",
+      "openPage": "should be {String|Array} (https://webpack.js.org/configuration/dev-server/#devserveropenpage)",
```
Comment (hiroppy) [inline lib/options.json]: We have to update the documentation.

File: lib/utils/runOpen.js [hunk 5]
```diff
@@ -13,16 +13,22 @@ function runOpen(uri, options, log) {
...
-      `${openMessage}. If you are running in a headless environment, please do not use the --open flag`
-    );
-  });
+  return Promise.all(
+    pages.map((page) => {
+      const pageUrl = page && isAbsoluteUrl(page) ? page : `${uri}${page}`;
+
+      return open(pageUrl, openOptions).catch(() => {
+        log.warn(
+          `${openMessage}. If you are running in a headless environment, please do not use the --open flag`
```
Comment (hiroppy) [inline lib/utils/runOpen.js]: I think this log is very noisy when logs are many. What do you think?

Comment (evilebottnawi) [reply lib/utils/runOpen.js]: I think it is normal, something can be happens and we should output warning about this

Comment (hiroppy) [reply lib/utils/runOpen.js]: Outputted messages are always the same. So it is better to put the output together or add a specific word to the message.

Comment (evilebottnawi) [reply lib/utils/runOpen.js]: Yes, you are right, let's insert open page in error message

File: lib/options.json (hunk 4 above)
Comment (digitaljohn) [reply lib/options.json]: I have opened a PR.
https://github.com/webpack/webpack.js.org/pull/3306

File: lib/utils/runOpen.js (hunk 5 above)
Comment (digitaljohn) [reply lib/utils/runOpen.js]: @hiroppy @evilebottnawi I have made adjustments. Is this what you had in mind?

File: lib/utils/runOpen.js [hunk 6]
```diff
@@ -24,7 +24,7 @@ function runOpen(uri, options, log) {
 
       return open(pageUrl, openOptions).catch(() => {
         log.warn(
-          `${openMessage}. If you are running in a headless environment, please do not use the --open flag`
+          `Unable to open ${pageUrl} in browser${openOptionValue}. If you are running in a headless environment, please do not use the --open flag`
```
Comment (evilebottnawi) [inline lib/utils/runOpen.js]: `: ${options.open}` -> `: "${options.open}"`

`Unable to open ${pageUrl} in browser${openOptionValue}` -> `Unable to open "${pageUrl}" in browser${openOptionValue}`

Comment (digitaljohn) [reply lib/utils/runOpen.js]: Updated! Thanks.

**Comment by digitaljohn** [comment]
@hiroppy @evilebottnawi any ideas on the failing tests?

**Comment by evilebottnawi** [comment]
@digitaljohn sometimes chromeless failed on windows due unknown reason, so CI failed, don't worry

**Pull request merged**